File: ruby/dni_ruc.rb
require 'net/http'
require 'json'

TOKEN = '<token>'
BASE_URL = URI('https://api.json.pe/api/dni')

uri = BASE_URL
request = Net::HTTP::Post.new(uri)
request['Authorization'] = "Bearer #{TOKEN}"
request['Content-Type'] = 'application/json'
request.body = JSON.generate(dni: '27427864')

response = Net::HTTP.start(uri.hostname, uri.port, use_ssl: true) do |http|
  http.request(request)
end

raise "Error #{response.code}: #{response.body}" unless response.is_a?(Net::HTTPSuccess)

puts response.body

File: dart/dni_ruc.dart
import 'dart:convert';
import 'dart:io';

const token = '<token>';
const url = 'https://api.json.pe/api/dni';

Future<void> main() async {
  final client = HttpClient();
  final request = await client.postUrl(Uri.parse(url));
  request.headers.set(HttpHeaders.authorizationHeader, 'Bearer $token');
  request.headers.set(HttpHeaders.contentTypeHeader, 'application/json');
  request.add(utf8.encode(jsonEncode({'dni': '27427864'})));

  final response = await request.close();
  final body = await utf8.decoder.bind(response).join();

  if (response.statusCode >= 400) {
    throw HttpException('Error ${response.statusCode}: $body');
  }

  print(body);
}
